homeassistant/components/tado/climate.py

```diff
@@ -118,80 +103,88 @@ def create_climate_device(tado, hass, zone, name, zone_id):
         return
 
     min_temp = float(temperatures["celsius"]["min"])
     max_temp = float(temperatures["celsius"]["max"])
     step = temperatures["celsius"].get("step", PRECISION_TENTHS)
 
-    data_id = f"zone {name} {zone_id}"
-    device = TadoClimate(
+    entity = TadoClimate(
         tado,
         name,
         zone_id,
-        data_id,
-        hass.config.units.temperature(min_temp, unit),
-        hass.config.units.temperature(max_temp, unit),
+        zone_type,
+        min_temp,
+        max_temp,
         step,
-        ac_device,
-        hot_water_device,
         ac_support_heat,
+        fallback,
     )
-
-    tado.add_sensor(
-        data_id, {"id": zone_id, "zone": zone, "name": name, "climate": device}
-    )
-
-    return device
+    return entity
 
 
 class TadoClimate(ClimateDevice):
     """Representation of a Tado climate device."""
 
     def __init__(
         self,
-        store,
+        tado,
         zone_name,
         zone_id,
-        data_id,
+        zone_type,
         min_temp,
         max_temp,
         step,
-        ac_device,
-        hot_water_device,
         ac_support_heat,
-        tolerance=0.3,
+        fallback,
     ):
         """Initialize of Tado climate device."""
-        self._store = store
-        self._data_id = data_id
+        self._tado = tado
 
         self.zone_name = zone_name
         self.zone_id = zone_id
+        self.zone_type = zone_type
 
-        self._ac_device = ac_device
-        self._hot_water_device = hot_water_device
+        self._ac_device = zone_type == TYPE_AIR_CONDITIONING
         self._ac_support_heat = ac_support_heat
         self._cooling = False
 
         self._active = False
         self._device_is_active = False
 
-        self._unit = TEMP_CELSIUS
         self._cur_temp = None
         self._cur_humidity = None
         self._is_away = False
         self._min_temp = min_temp
         self._max_temp = max_temp
         self._step = step
         self._target_temp = None
-        self._tolerance = tolerance
+
+        if fallback:
+            _LOGGER.debug("Default overlay is set to TADO MODE")
+            # Fallback to Smart Schedule at next Schedule switch
+            self._default_overlay = CONST_OVERLAY_TADO_MODE
+        else:
+            _LOGGER.debug("Default overlay is set to MANUAL MODE")
+            # Don't fallback to Smart Schedule, but keep in manual mode
+            self._default_overlay = CONST_OVERLAY_MANUAL
 
         self._current_fan = CONST_MODE_OFF
         self._current_operation = CONST_MODE_SMART_SCHEDULE
         self._overlay_mode = CONST_MODE_SMART_SCHEDULE
 
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+        _LOGGER.debug("Registering for dispatcher updates for zone %d", self.zone_id)
+        async_dispatcher_connect(
+            self.hass,
+            SIGNAL_TADO_UPDATE_RECEIVED.format("zone", self.zone_id),
+            self.update,
```

Comment:
We should make an intermediate function that we use for target here:
```py
@callback
def async_update_callback():
    """Schedule an entity update."""
    self.async_schedule_update_ha_state(True)

async_dispatcher_connect(
    self.hass,
    SIGNAL_TADO_UPDATE_RECEIVED.format("zone", self.zone_id),
    self.async_update_callback,
)
```

Sorry that I forgot about this earlier.

Reply:
Thanks!